src/rest_vol.c

```diff
@@ -732,6 +796,350 @@ H5_rest_set_connection_information(void)
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5_rest_authenticate_with_AD(H5_rest_ad_info_t *ad_info)
+{
+    const char *access_token_keys[] = { "access_token", (const char *) 0 };
+    const char *refresh_token_keys[] = { "refresh_token", (const char *) 0 };
+    const char *expires_in_key[] = { "expires_in", (const char *) 0 };
+    const char *token_cfg_file_name = ".hstokencfg";
+    yajl_val    parse_tree = NULL, key_obj = NULL;
+    size_t      token_cfg_file_pathname_len = 0;
+    FILE       *token_cfg_file = NULL;
+    char       *token_cfg_file_pathname = NULL;
+    char       *home_dir = NULL;
+    char       *access_token = NULL;
+    char       *refresh_token = NULL;
+    time_t      token_expires = -1;
+    char        tenant_string[1024];
+    char        data_string[1024];
+    herr_t      ret_value = SUCCEED;
+
+    if (!ad_info)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "Active Directory info structure is NULL");
+
+    /* Set up miscellaneous options */
+    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_HTTPAUTH, CURLAUTH_BEARER))
+        FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTSET, FAIL, "can't set HTTP authentication method: %s", curl_err_buf);
+    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L))
+        FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTSET, FAIL, "can't set cURL FOLLOWLOCATION opt.: %s", curl_err_buf);
+    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_DEFAULT_PROTOCOL, "https"))
+        FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTSET, FAIL, "can't set default protocol to HTTPS: %s", curl_err_buf);
+
+    curl_headers = curl_slist_append(curl_headers, "Content-Type: application/x-www-form-urlencoded");
+
+    if (CURLE_OK != curl_easy_setopt(curl, CURLOPT_HTTPHEADER, curl_headers))
+        FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTSET, FAIL, "can't set cURL HTTP headers: %s", curl_err_buf);
+
+    /* Check if an access token config file exists in the user's home directory */
+#ifdef WIN32
+    {
+        char *home_drive = NULL;
+
+        if (NULL == (home_drive = getenv("HOMEDRIVE")))
+            FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTGET, FAIL, "reading access token config file - unable to retrieve location of home directory");
+
+        if (NULL == (home_dir = getenv("HOMEPATH")))
+            FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTGET, FAIL, "reading access token config file - unable to retrieve location of home directory");
+
+        token_cfg_file_pathname_len = strlen(home_drive) + strlen(home_dir) + strlen(token_cfg_file_name) + 3;
+        if (NULL == (token_cfg_file_pathname = RV_malloc(token_cfg_file_pathname_len)))
+            FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTALLOC, FAIL, "unable to allocate space for access token config file pathname");
+
+        if (snprintf(token_cfg_file_pathname, token_cfg_file_pathname_len, "%s\%s\%s", home_drive, home_dir, token_cfg_file_name) < 0)
```

Comment:
Looks like I left a missing extra '\' character to escape the '\' in the output string. I know this same code exists in the .hscfg file parsing logic, so I wonder how that works correctly for Chevron on Windows..

Reply:
Fixed in both places.

Reply:
Great, thanks!